packages/cli/src/rpc/navie/services/threadIndexService.ts

```diff
@@ -63,11 +71,64 @@ export interface ThreadIndexItem {
 @singleton()
 @injectable()
 export class ThreadIndexService {
+  static readonly MIGRATION_RETRIES = 5;
   static readonly DEFAULT_DATABASE_PATH = join(homedir(), '.appmap', 'navie', 'thread-index.db');
   static readonly DATABASE = 'ThreadIndexDatabase';
 
   constructor(@inject(ThreadIndexService.DATABASE) private readonly db: sqlite3.Database) {
-    this.db.exec(INITIALIZE_SQL);
+    this.db.exec(INITIALIZE_SESSION_SQL);
+  }
+
+  private shouldMigrate(): boolean {
+    const currentVersion = this.db.get('PRAGMA user_version')?.user_version;
+    return currentVersion !== SCHEMA_VERSION;
+  }
+
+  /**
+   * Migrates the database schema to the latest version. This should be called upon application
+   * initialization to ensure that the database is up to date.
+   */
+  public async migrate() {
+    if (!this.shouldMigrate()) return;
+
+    let lockRelease: (() => Promise<void>) | undefined;
+    try {
+      lockRelease = await lock(ThreadIndexService.DEFAULT_DATABASE_PATH, {
+        retries: ThreadIndexService.MIGRATION_RETRIES,
+        stale: 10000,
+        lockfilePath: `${ThreadIndexService.DEFAULT_DATABASE_PATH}.migration.lock`,
+      });
+
+      if (!this.shouldMigrate()) {
+        console.info('ThreadIndexService: migration completed by another process');
+        return;
+      }
+
+      console.info(`ThreadIndexService: migrating schema to v${SCHEMA_VERSION}`);
+      try {
+        this.db.exec('PRAGMA journal_mode = WAL');
+        this.db.exec('BEGIN TRANSACTION');
+        this.db.exec(INITIALIZE_DB_SQL);
```

Comment:
This isn't fully future-proof. We'll need to change this a bit if we ever need to mutate the schema in a non-idempotent manner, but that's an issue for another day.